src/github.com/ebay/beam/util/bytes/b.go

```diff
@@ -26,6 +26,19 @@ func Copy(src []byte) []byte {
 	return r
 }
 
+// Concat returns a new byte slice with all the parts appended together.
+func Concat(parts ...[]byte) []byte {
+	total := 0
+	for _, p := range parts {
+		total += len(p)
+	}
+	out := make([]byte, 0, total)
+	for _, p := range parts {
+		out = append(out, p...)
+	}
+	return out
```

Comment:
I think this could be simplified to `bytes.Join(parts, []byte{})` (using package `bytes` from the standard library). At that point, it might be too shallow to export from this package--your call.

Reply:
Good call, hadn't seen that.

Reply:
I removed bytes.Concat and added a concat helper in keys_test.go, because otherwise wrapping everything in [][]byte{ ... } makes it even more unreadable.